design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+    telephone: String! @doc(description: "The telephone number")
+    street: [String]! @doc(description: "An array of strings that define the street number and name")
+    city: String! @doc(description: "The city or town")
+    region: String! @doc(description: "The region name")
+    region_id: Int @doc(description: "The region ID")
+    postcode: String! @doc(description: "The customer's ZIP or postal code")
+    country_id: String! @doc(description: "Deprecated: use `country_code` instead.")
+}
+
+type EstimateShippingCostForNegotiableQuoteOutput {
+    shipping_methods: [ShippingMethodsOutput]! @doc(description: "All shipping methods for the shipping address")
+}
+
+type EstimateShippingCostByAddressIdOutput {
+    shipping_methods: [ShippingMethodsOutput]! @doc(description: "All shipping methods for the shipping address")
+}
+
+type ShippingMethodsOutput {
+    carrier_code: String @doc(description: "Shipping carrier code")
+    method_code: String @doc(description: "Shipping method code")
+    carrier_title: String @doc(description: "Shiping carrier title")
+    method_title: String @doc(description: "Shipping method title")
+    amount: String @doc(description: "Shipping amount")
+    base_amount: String @doc(description: "Shipping base amount")
+    available: String @doc(description: "Shipping method availble")
+    error_message: String @doc(description: "Shipping method error message")
+    price_excl_tax: String @doc(description: "Shipping method price with exclusive tax")
+    price_incl_tax: String @doc(description: "Shipping method price with inclusive tax")
+}
+
+input NegotiableQuoteEstimateShippinAddressIdInput {
+    cart_id: String! @doc(description: "Cart hash")
+    address_id: String! @doc(description: "Masked shipping address ID assigned to negotiable quote")
+}
+
+input NegotiableQuoteShippingInformationInput {
+    cart_id: String! @doc(description: "Cart hash")
+    address_info: NegotiableQuoteAddressInformationInput @doc(description: "Input for address and shipping info")
+}
+
+input NegotiableQuoteAddressInformationInput {
+    shipping_address: NegotiableQuoteShipAddressInput @doc(description: "Shipping address assigned to negotiable quote")
+    billing_address: NegotiableQuoteBillAddressInput @doc(description: "Billing address assigned to negotiable quote")
+    shipping_carrier_code: String! @doc(description: "Shipping carrier code")
+    shipping_method_code: String! @doc(description: "Shipping method code")
+}
+
+input NegotiableQuoteBillAddressInput {
+    firstname: String! @doc(description: "The first name of the person associated with the billing address")
+    lastname: String! @doc(description: "The family name of the person associated with the billing address")
+    telephone: String! @doc(description: "The telephone number")
+    street: [String]! @doc(description: "An array of strings that define the street number and name")
```

Comment:
```suggestion
    street: [String]! @doc(description: "A list of strings that define the street number and name")
```

Depending on the language a client is written in, might use a different datatype.